### PR TITLE
DG: Add Use Cases

### PR DESCRIPTION
**Pull request by S-K-Y-Light**
Use Cases for Students, Consultations & Lesson Commands.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -478,317 +478,330 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1. User requests to add a consultation by providing the necessary details (date, time).
+2. TAHub validates the inputs.
+3. TAHub adds the consultation with the provided details.
+4. TAHub displays the updated consultation list.
+<br>
+Use case ends.
 
 **Extensions**
+* 2a. One or more input parameters are missing or invalid.
+    * 1a1. TAHub shows an error message indicating the missing or invalid fields.
```
Comment (notnotmax) [inline docs/DeveloperGuide.md]: I think you meant to write 2a1.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -478,317 +478,330 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+<br>
+Use case ends.
 
 **Extensions**
+* 2a. One or more input parameters are missing or invalid.
+    * 1a1. TAHub shows an error message indicating the missing or invalid fields. 
+    <br>
+    Use case ends
+* 2b. Duplicate consultation Exists (Consultation date & time matches an existing consultation)
+    * 1b1. TAHub shows a duplicate error message.
```
Comment (notnotmax) [inline docs/DeveloperGuide.md]: 2b1?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -478,317 +478,330 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-**MSS**
-
-1. TA lists existing consultations
-2. TA selects consultation by index
-3. TA specifies students to remove by name
-4. System validates student existence in consultation
-5. System removes specified students from consultation 
-6. Success message shown
+**MSS:**
+1. User requests to remove a specific student from the consultation by providing the necessary details (Consultation Index, Student Index, Student Name)
```
Comment (notnotmax) [inline docs/DeveloperGuide.md]: Correct me if I'm wrong but I don't think `removefromconsult` allows deleting a student by index, only their full name.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -478,317 +478,330 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+2. TAHub validates filename.
+3. TAHub creates CSV file with consultation data.
+4. TAHub copies file to home directory.
+5. Success message shows number of consultations exported.
+<br>
+Use case ends.
 
-**MSS**
+**Extensions:**
+* [Same extensions as UC6]
```
Comment (notnotmax) [inline docs/DeveloperGuide.md:749]: No idea if this is allowed, but well done if it is.

**Review by notnotmax — CHANGES_REQUESTED** [review]
Looks good so far, pending minor corrections.

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (S-K-Y-Light) [reply docs/DeveloperGuide.md:749]: This was done by @yhanyi. I've no idea either, but I left it as it is.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (S-K-Y-Light) [reply docs/DeveloperGuide.md]: You're correct. Will Modify accordingly.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (S-K-Y-Light) [reply docs/DeveloperGuide.md]: Yep. Good catch.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (S-K-Y-Light) [reply docs/DeveloperGuide.md]: Yep. Good catch

**Review by notnotmax — APPROVED** [review]

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -797,9 +797,167 @@ Use case ends.
...
+
+**<u>Use case: UC21 - Mark Student Participation in Lesson</u>**
+
+**Precondition:** The Student & Lesson exists in TAHub. Student is already in the Lesson.
+
+**MSS:**
+1. User requests to mark a specific student's participation in the lesson by providing the necessary details (Lesson Index, Student Name, Participation)
+2. TAHub validates the inputs.
+3. TAHub marks the student's participation in the lesson.
+4. TAHub displays the updated student attendance.
```
Comment (notnotmax) [inline docs/DeveloperGuide.md]: Looks like you forgot to modify this after the copy paste?

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -797,9 +797,167 @@ Use case ends.
...
+   <br>
+   Use case ends.
```
Comment (notnotmax) [inline docs/DeveloperGuide.md:934]: Also, you may want to add as an extension that if the participation is set to >0, the student's attendance is also set to true.

**Review by notnotmax — COMMENTED** [review]
LGTM, minor errors.

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (S-K-Y-Light) [reply docs/DeveloperGuide.md]: Yep. Resolved.

File: docs/DeveloperGuide.md (hunk 6 above)
Comment (S-K-Y-Light) [reply docs/DeveloperGuide.md:934]: Will Do!

**Pull request merged**